### DISCUSSION TITLE
Maven nature added automatically

### DISCUSSION THREAD

**Discussion by lwrage** [eclipse-m2e/m2e-core#1395] [Q&A]
I have been developing eclipse plugins that are built using tycho for several years. All the sudden I have new and  unexpected behavior in my Eclipse development environment: a maven nature is added automatically to a set of plugin projects. If I remove the changes, the maven nature appears again after a while. This started happening a couple of days ago, without me installing any updates. This is with Eclipse 2022-06 that includes M2E 1.20.1, so I assume m2e is doing this.
Note that this does not happen for all projects that have a pom.xml file, but just for a set of recently added projects that are built independently from the rest of the projects in the workspace.
Does anybody know what may be triggering the automatic addition of a maven nature, and if there is a way to turn it off?

**Comment by laeubi** [comment]
> Eclipse 2022-06 that includes M2E 1.20.1

Is really outdated, so I suspect it will be hard to get any sensible answers, in general it might be you installed a connector that is configuring the project but usually all m2e related stuff is disabled when the nature is not present.

**Comment by lwrage** [comment]
I'm in the process of upgrading to Eclipse 2023-03, so I hope this won't happen there. The weird thing is that I didn't install anything new in the last couple of months.

**Reply by laeubi** [reply]
At least I can't think about any case where suddenly the behavior drastically changes, so even if you don't installed anything you maybe have changed the pom (what is triggering a plugin).
